src/ipc/ipc4/logging.c

```diff
@@ -198,6 +199,12 @@ int ipc4_logging_enable_logs(bool first_block,
 
 #endif
 
+#if CONFIG_LIBRARY
+int ipc4_logging_shutdown(void)
+{
+	return 0;
+}
```

Comment:
You might be able to just use the no-op version of ipc4_logging_enable_logs() for CONFIG_LIBRARY.